scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryScioContext.scala

```diff
@@ -85,8 +114,16 @@ final class BigQueryScioContext(@transient val self: ScioContext) extends Serial
    * behavior, start the query string with `#legacysql` or `#standardsql`.
    */
   def typedBigQuery[T <: HasAnnotation: ClassTag: TypeTag: Coder](
-    newSource: String = null): SCollection[T] =
-    self.read(BigQueryTyped.dynamic[T](newSource))
+    newSource: String = null): SCollection[T] = {
+    val bqt = BigQueryType[T]
+    if (bqt.isStorage) {
```

Comment:
:art:

```scala
val io = if {} else {}

self.read(io)
```

Reply:
Can't really do that because one branch has `params` and the other doesn't.

Reply:
oh yes! sorry missed that.